lib/Runtime/ByteCode/ByteCodeSerializer.cpp

```diff
@@ -4543,7 +4543,7 @@ class ByteCodeBufferReader
         auto propertyCount = serialized->propertyCount;
         auto extraSlotCount = serialized->extraSlots;
 
-        Assert(serialized->offset + sizeof(PropertyIdArray) < deserializeInto->GetLength());
+        Assert(serialized->offset + sizeof(PropertyIdArray) < deserializeInto->GetLength() || propertyCount == 0);
```

Comment:
Are you sure that this is valid? Even if `propertyCount` is 0 the next thing we do is go and interpret `serialized->offset + deserializeInto->GetBuffer()` as a `PropertyIdArray` and go and write things on it; if this assert is failing, that suggests that we are writing into the memory of something else.